SteamTogether.Scraper/Worker.cs

```diff
@@ -27,17 +27,22 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             await scraper.RunSync();
         }
 
-        var schedule = NCrontab.CrontabSchedule.Parse(opts.Schedule);
+        _logger.LogInformation("Using schedule: {Schedule}", opts.Schedule);
+        var cron = CronExpression.Parse(opts.Schedule);
         while (!stoppingToken.IsCancellationRequested)
         {
-            var dateTimeService = _serviceProvider.GetRequiredService<IDateTimeService>();
-            var now = dateTimeService.GetCurrentTime();
+            var utcNow = DateTime.UtcNow;
+            var utcNext = cron.GetNextOccurrence(utcNow);
 
-            var nextExecutionTime = schedule.GetNextOccurrence(dateTimeService.GetCurrentTime());
-            using var timer = new PeriodicTimer(nextExecutionTime - now);
+            if (utcNext == null)
+            {
+                _logger.LogWarning("No next run found, stopping worker");
+                break;
+            }
 
-            _logger.LogInformation("Next worker run: {Next}", nextExecutionTime);
-            await timer.WaitForNextTickAsync(stoppingToken);
```

Comment:
What is the problem with PeriodicTimer? 
It is a new feature of .net6 and should help to avoid potetial memory leaks
https://www.ilkayilknur.com/a-new-modern-timer-api-in-dotnet-6-periodictimer

Reply:
In the article, it is said that memory leak issues are related to using a callback function. It is not the case for the `Task.Delay`.

There is no problem with the `PeriodicTimer`. It is just less suitable in this situation. We want to wait for some time inside the loop, but, as far as I understand, the primary purpose of `PeriodicTimer` is to act like a periodic timer, not like a one-time-run timer.

Also, the creation of a new object on every iteration seems unnecessary.

Overall, the `Task.Delay` usage looks more concise.

Reply:
Decided to use `Task.Delay` for now.